### ISSUE TITLE
Animation panel: curve editor: add zoom to fit command and shortcut

### ISSUE THREAD

**Issue by hsandt** [godotengine/godot-proposals#7547]
### Describe the project you are working on

A combat platformer

### Describe the problem or limitation you are having in your project

It's tedious to scale the Animation curve editor to fit all key points, esp. vertically. I need to use Ctrl+Alt+Mouse wheel scroll to zoom in, scroll to move and adjust offset, then zoom in/out again, until I get the correct range.

For rotation which uses radian angles, it's particularly striking as the default range is huge for these numbers (which are below 3.14).

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a right-click > Zoom to fit, possibly with variants: horizontally, vertically, both so user can preserve the range along X or Y if they want.

It could also have a button and shortcut equivalent.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![2023-08-23 Animation curve zoom to fit - suggested command and button](https://github.com/godotengine/godot-proposals/assets/1067772/4027af6e-3f15-4963-a105-e6fdd327e727)

You can change the command/button placement to a more fitting location if needed

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it's a core window

### Is there a reason why this should be core and not an add-on in the asset library?

Yes, it's a core window

**Comment by TokisanGames** [comment]
Yes, [this is the comment ](https://github.com/godotengine/godot-proposals/issues/3141#issuecomment-898944531) where I suggested many needed features that I now realize did not get implemented. The biggest thing the bezier editor desperately needs is auto scale based on the current track.

When animating, multiple tracks need to be edited simultaneously. Since most tracks are at different scales (eg. position is -1000 to 1000 while rotation is 0-6), it is exceedingly difficult to edit the curves and requires a lot of unnecessary operations between editing.

> I'm not sure if they meant actually auto-fit without user action, but let's start with a manual Zoom to Fit.

* There should be an autofit option that can be enabled. After that, any time a track is selected, the window zooms to encompass that track. 
* Having variations such as full or vertical only is a good idea. 
* Having a 'F' key to manually fit around the current track when this auto-fit is disabled is also a good idea. 

> Add a right-click > Zoom to fit...

While having a menu option is fine, it most definitely should not be the only option. When animating, we want less clicks not more. No 1 is current. This suggestion is No 2. No 3. is what an animator needs.
1. Click a track, zoom way out, zoom way in, edit, click a track, zoom way out, zoom way in, edit
2. Click a track, right click, zoom to fit, edit, click a track, right click, zoom to fit, edit
3. Click a track, edit, click a track, edit

**Comment by viksl** [comment]
Just wanted to mention this in case someone missed it since things have moved forward I guess.
There's a Fit To Panel button which fits everything to your panel, it also has a shortcut (same name)
The F to focus on selected frames was also implemented.
